rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/InjectionUtils.java

```diff
@@ -1167,10 +1161,12 @@ public static void injectContextMethods(Object requestObject,
                 if (!cri.isSingleton()) {
                     InjectionUtils.injectThroughMethod(requestObject, method, o, message);
                 } else {
-                    ThreadLocalProxy<Object> proxy 
-                        = (ThreadLocalProxy<Object>)cri.getContextSetterProxy(method);
+                    Object proxy = extractFromSetter(requestObject, method);
+                    if (!(proxy instanceof ThreadLocalProxy)) {
```

Comment:
Hi, can you explain please why this check is done, what else can the proxy be, apart from null or ThreadLocalProxy ? Thanks